modules/drivers/mongo/test/metabase/driver/mongo/util_test.clj

```diff
@@ -0,0 +1,13 @@
+(ns metabase.driver.mongo.util-test
+  (:require
+   [clojure.test :refer :all]
+   [metabase.driver.mongo.connection :as mongo.connection]
+   [metabase.driver.mongo.util :as mongo.util]
+   [metabase.test :as mt]))
+
+(deftest do-find-returns-lazy-seq-test
+  (mt/test-driver :mongo
+    (testing "do-find returns a lazy seq #42133"
+      (mongo.connection/with-mongo-database [db (mt/db)]
+        (is (= clojure.lang.LazySeq
+               (type (mongo.util/do-find (mongo.util/collection db "venues")))))))))
```

Comment:
Rather than coding to a concrete type, and assuming that it hasn't been realized, you could instead assert that `realized?` is false.

Reply:
oh yea that'd be nicer.